js/common/modules/@arangodb/foxx/check-args.js

```diff
@@ -46,6 +48,39 @@ function normalizeMimeType (mime) {
   ]));
 }
 
+function isJoiSchema (value, expectOpaque) {
+  if (!value) return false;
+  if (value.isJoi) return true;
+  if (expectOpaque) return false;
+  if (typeof value !== "object") return false;
+  if (Array.isArray(value)) return false;
+  const values = Object.values(value);
+  if (!values.length) return false;
+  return values.every(value => isJoiSchema(value, true));
+}
+
+function createSchemaValidator (schema) {
+  if (schema.isJoi) {
+    return (value) => schema.validate(value);
+  }
+  if (schema === false) {
+    return (value) => {
```

Comment:
Technically AJV allows passing `false` as a schema to generate a validator that doesn't match anything but that also doesn't match _nothing_ (i.e. `undefined`, `null` or the empty string). So we intercept that special case to generate a validation method that converts "nothing" to `undefined` for consistency but otherwise rejects all input.